src/App.tsx

```diff
@@ -1,61 +1,107 @@
+import { useState, ChangeEvent, FormEvent } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [todoUser, setTodoUser] = useState<string>('0');
+  const [todos, setTodos] = useState(todosFromServer);
+  const [userError, setUserError] = useState<boolean>(false);
+  const [titleError, setTitleError] = useState<boolean>(false);
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setTodoUser(event.target.value);
+    setUserError(false);
+  };
+
+  const handleCreateTodo = (event: FormEvent) => {
+    if (todoUser === '0') {
+      setUserError(true);
+    }
+
+    if (todoTitle.length === 0) {
+      setTitleError(true);
+    }
+
+    event.preventDefault();
+    const newId
+      = todos.reduce(
+        (acc, curr) => (curr.id > acc.id ? curr : acc), todos[0],
+      ).id + 1;
```

Comment:
i would sort and then find the biggest id, but i guess that also makes sense (but your approach may be less readable for other people at the project)